p2pd/main.go

```diff
@@ -112,6 +124,20 @@ func main() {
 		log.Fatal(err)
 	}
 
+	if *autonat {
+		var opts []libp2p.Option
+		if *QUIC {
```

Comment:
Why is this special casing necessary for QUIC? Could we add a comment here, and also [here](https://github.com/libp2p/go-libp2p-daemon/pull/64/files#diff-f3664e75b1c754c40158dd95808a9c3dR83)?

Reply:
That's for the autonat service being able to dial QUIC addrs.

Reply:
added comments.